File: static/js/rellenarTb.js
$(document).ready(function() {
  $("#search, #select").on("input", function() {
    var valor_seacrh = $("#search").val();
    var valor_select = $("#select").val();
    $.ajax({
      type: "POST",
      url: "/ficha",
      data: {
        valor_seacrh: valor_seacrh, 
        valor_select: valor_select
      },
      success: function(data) {
        var html = ""; // Variable para almacenar el contenido HTML de la tabla
        data.forEach(function(socio) {
          // Construir una fila HTML con los datos del socio
          html += "<tr>";
          html += "<td>" + socio.campo1 + "</td>";
          html += "<td>" + socio.campo2 + "</td>";
          html += "<td>" + socio.campo3 + "</td>";
          html += "<td>" + socio.campo4 + "</td>";
          html += "<td>" + socio.campo5 + "</td>";
          html += "<td>" + socio.campo6 + "</td>";
          html += "<td>" + socio.campo7 + "</td>";
          html += "<td>" + socio.campo8 + "</td>";
          html += "<td>" + socio.campo9 + "</td>";
          html += "<td>" + socio.campo10 + "</td>";
          html += "<td>" + socio.campo11 + "</td>";
          html += "<td>" + socio.campo12 + "</td>";
          html += "<td>" + socio.campo13 + "</td>";
          html += "<td>" + socio.campo14 + "</td>";
          html += "<td>" + socio.campo15 + "</td>";
          html += "<td>" + socio.campo16 + "</td>";
          html += "<td>" + socio.campo17 + "</td>";
          html += "<td>" + socio.campo18 + "</td>";
          html += "<td>" + socio.campo19 + "</td>";
          html += "<td>" + socio.campo20 + "</td>";
          html += "<td>" + socio.campo21 + "</td>";
          html += "<td>" + socio.campo22 + "</td>";
          html += "<td>" + socio.campo23 + "</td>";
        });
        // Establecer el contenido HTML completo en la tabla
        $("#table tbody").html(html);
      }
    });
  });
});

File: models/rellenarTB.py
import mysql.connector
from flask import Blueprint
from flask import render_template,request,jsonify
from flask_login import login_required


ruta1_bp = Blueprint('ruta1', __name__)

@ruta1_bp.route("/ficha")
def index():
    return render_template("ficha.html")

@ruta1_bp.route('/ficha', methods=['GET', 'POST'])
@login_required
def fichas():
    mydb = mysql.connector.connect(
    host="186.158.11.37",
    user="mysql_user",
    password="Aedr15150302",
    database="database"
    )
    busqueda = request.form.get('valor_seacrh')
    valor_ficha = request.form.get('valor_select')
    mycursor = mydb.cursor()
    if valor_ficha == None or valor_ficha == 'todos':
        if busqueda:
            query = "SELECT dni, apellido, nombre, domicilio, telefono, fechanac, caracter,fechaingreso, fechaegreso, observacion, tipo, categoriaJug, dniJug, apellidoJug, nombreJug, fechanaciJug, tipoJug2, categoriaJug2, dniJug2, apellidoJug2, nombreJug2, fechanaciJug2 FROM socios WHERE nombre LIKE %s"
            mycursor.execute(query, ("%" + busqueda + "%",))
        else: 
            query = "SELECT dni, apellido, nombre, domicilio, telefono, fechanac, caracter, fechaingreso, fechaegreso, observacion, tipo, categoriaJug, dniJug, apellidoJug, nombreJug, fechanaciJug, tipoJug2, categoriaJug2, dniJug2, apellidoJug2, nombreJug2, fechanaciJug2 FROM socios"
            mycursor.execute(query)
    else:     
        if busqueda:
            query = "SELECT dni, apellido, nombre, domicilio, telefono, fechanac, caracter,fechaingreso, fechaegreso, observacion, tipo, categoriaJug, dniJug, apellidoJug, nombreJug, fechanaciJug, tipoJug2, categoriaJug2, dniJug2, apellidoJug2, nombreJug2, fechanaciJug2 FROM socios WHERE nombre LIKE %s and tipo LIKE %s"
            mycursor.execute(query, ("%" + busqueda + "%", "%" + valor_ficha + "%"))
        else:
            query = "SELECT dni, apellido, nombre, domicilio, telefono, fechanac, caracter,fechaingreso, fechaegreso, observacion, tipo, categoriaJug, dniJug, apellidoJug, nombreJug, fechanaciJug, tipoJug2, categoriaJug2, dniJug2, apellidoJug2, nombreJug2, fechanaciJug2 FROM socios WHERE tipo LIKE %s"
            mycursor.execute(query,("%" + valor_ficha + "%",))

    resultados = mycursor.fetchall()
    
    datos = []
    for fila in resultados:
        diccionario = {
            "campo1": fila[0],
            "campo2": fila[1],
            "campo3": fila[2],
            "campo4": fila[3],
            "campo5": fila[4],
            "campo6": fila[5],
            "campo7": fila[6],
            "campo8": fila[7],
            "campo9": fila[8],
            "campo10": fila[9],
            "campo11": fila[10],
            "campo12": fila[11],
            "campo13": fila[12],
            "campo14": fila[13],
            "campo15": fila[14],
            "campo16": fila[15],
            "campo17": fila[16],
            "campo18": fila[17],
            "campo19": fila[18],
            "campo20": fila[19],
            "campo21": fila[20],
            "campo22": fila[21],
        }
        datos.append(diccionario)
        datos=datos
    return jsonify(datos)
